casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -157,6 +157,20 @@ object Validate {
       true.pure[F]
     }
 
+  def version[F[_]: Applicative: Log](b: BlockMessage, version: Long): F[Boolean] = {
+    val blockVersion = b.header.get.version
```

Comment:
I don't really like the unsafe `b.header.get` here. I understand that the blocks should always have a header but still it feels bad. 

I think it's safe to say that the version validation fails if the block header is missing.